File: bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/document/Frame.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
///**
//* Utility class for HTML <code>IFrame</code> element.
//*/
//vjo.ctype("org.eclipse.vjet.vsf.document.Frame")
//.needs(["org.eclipse.vjet.vsf.Element",
//	"org.eclipse.vjet.vsf.client.Browser"])
//.props({
//	/**
//	* Resizes iframe's size.
//	*
//	* @param {String} iframeId 
//	*        the id of the iframe to be resized
//	*/
//	//> public void resize(String);
//	resize : function(psIframeId) {
//		var E = org.eclipse.vjet.vsf.Element, f;
//		f = E.get(psIframeId);
//		if(f && typeof(f.document)!="unknown")
//		{
//			var oDoc=f.document ? f.document : f.contentDocument, 
//					db = oDoc.body,
//					es = f.style,
//					c = org.eclipse.vjet.vsf.client.Browser,
//					w = "100%",
//					h = db.offsetHeight,
//					oh;
//			if(c.bSafari)
//			{
//				oh = db.offsetHeight;
//				w = oDoc.width;
//				h = document.doctype !==null ? oDoc.height+15 : oDoc.height+1;
//	                
//			}
//			else if (c.bFirefox)
//			{
//				w =	oDoc.width;
//				h = oDoc.height;
//			}
//			else if(c.bWin || c.bOpera )
//			{
//				w = db.scrollWidth;
//				h = c.bNav && document.doctype !==null ? db.scrollHeight+30 : db.scrollHeight;
//			}
//			//es.width = w+"px";
//			es.height = h+"px";
//		}	
//	}
//})
//.endType();
//

File: vsf/bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/document/Frame.java
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
package org.eclipse.vjet.vsf.document;

import org.eclipse.vjet.dsf.javatojs.anno.AJsProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsProxy;
import org.mozilla.mod.javascript.Scriptable;
import org.eclipse.vjet.dsf.javatojs.anno.AExclude;
import org.eclipse.vjet.dsf.dap.proxy.INativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsTypeRef;
import org.eclipse.vjet.dsf.javatojs.anno.AJavaOnly;

//NativeJsProxy for vjo.dsf.document.Frame.js
@org.eclipse.vjet.dsf.resource.utils.CodeGen("NativeJsProxyGenerator")
@AJsProxy
public class Frame extends NativeJsProxy {

    /** for framework use only */
    @AExclude
    public Frame(Scriptable nativeObj){
        super(nativeObj);
    }

    /** internal use only */
    protected Frame(Object ...args){
        super(args);
    }

    public Frame() {
        super();
    }

    public static void resize(String psIframeId) {
        callStaticWithName("vjo.dsf.document.Frame", "resize", psIframeId);
    }

    @AJavaOnly
    public static final NativeJsTypeRef<Frame> prototype = NativeJsTypeRef.get(Frame.class);

    public static final INativeJsFuncProxy<NativeJsTypeRef<Frame>> resize = NativeJsFuncProxy.create(prototype, "resize");
}